grpc-cpp/plan.sh

```diff
@@ -0,0 +1,227 @@
+pkg_name=grpc-cpp
+pkg_distname=grpc
+pkg_origin=core
+pkg_version="1.19.1"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_source="https://github.com/grpc/grpc.git"
+pkg_shasum=noshasum
+
+pkg_deps=(
+    core/glibc
+    core/gcc-libs
+    core/zlib
+    core/openssl
+    core/protobuf
+    core/c-ares
+)
+pkg_build_deps=(
+    core/make
+    core/go
+    core/git
+    core/gcc
+    core/cmake
+    core/pkg-config
+    core/virtualenv
+    core/busybox-static
+)
+pkg_lib_dirs=(lib)
+pkg_include_dirs=(include)
+pkg_bin_dirs=(bin)
+pkg_description="A high performance, open-source universal RPC framework"
+pkg_upstream_url="https://grpc.io"
+
+do_begin() {
+  export HAB_ENV_CMAKE_FIND_ROOT_PATH_SEPARATOR=";"
+  export HAB_ENV_CMAKE_FIND_ROOT_PATH_TYPE="aggregate"
+}
+
+do_setup_environment() {
+  # making a cmake builddir that is the same as where the grpc tests expect it
+  set_buildtime_env BUILDDIR "cmake/build"
+
+  # this allows cmake users to utilize `CMAKE_FIND_ROOT_PATH` to find various cmake configs
+  push_runtime_env CMAKE_FIND_ROOT_PATH "${pkg_prefix}/lib/cmake/grpc"
+}
+
+do_download() {
+  REPO_PATH="$HAB_CACHE_SRC_PATH/$pkg_dirname"
+
+  # removing any previous git repo under the same package name that was downloaded
+  rm -rf "$REPO_PATH"
+
+  git clone "$pkg_source" "$REPO_PATH"
+
+  pushd "$REPO_PATH" || exit 1
+  git checkout "tags/v${pkg_version}"
+  git submodule init
+  git submodule update
+}
+
+do_clean() {
+  return 0
+}
+
+do_unpack() {
+  return 0
+}
+
+do_verify() {
+  return 0
+}
+
+do_prepare() {
+  mkdir -p "${BUILDDIR}"
+}
+
+do_build() {
+  ZLIB_PATH="$(pkg_path_for zlib)"
+  PROTOBUF_PATH="$(pkg_path_for protobuf)"
+
+  pushd "${BUILDDIR}" || exit 1
+  cmake \
+    -DCMAKE_FIND_ROOT_PATH="${CMAKE_FIND_ROOT_PATH}" \
+    -DgRPC_BUILD_TESTS="${DO_CHECK}" \
+    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+    -DgRPC_ZLIB_PROVIDER="package" \
+    -DZLIB_ROOT="${ZLIB_PATH}" \
+    -DgRPC_CARES_PROVIDER="package" \
+    -DgRPC_PROTOBUF_PROVIDER="package" \
+    -DProtobuf_INCLUDE_DIR="${PROTOBUF_PATH}/include" \
+    -DProtobuf_LIBRARY="${PROTOBUF_PATH}/lib/libprotobuf.so" \
+    -DProtobuf_PROTOC_LIBRARY="${PROTOBUF_PATH}/lib/libprotoc.so" \
+    -DgRPC_SSL_PROVIDER="package" \
+    -G "Unix Makefiles" \
+    ../..
+  make -j"$(nproc --ignore=1)"
+  popd || exit 1
+}
+
+do_check() {
+  testdir="${BUILDDIR}/testenv"
+  fulltestdir_path="$(realpath "$testdir")"
+
+  # clean up existing testenv
+  rm -rf "${testdir}"
+
+  # setup env
+  virtualenv "${testdir}"
+  source "${testdir}/bin/activate"
+
+  # add py deps
+  pip install six
+  pip install pyyaml
+  pip install twisted
+  pip install httplib2
+
+  # fix the interpreters without adding more runtime deps
+  sed -e "s#/usr/bin/env python.*#${fulltestdir_path}/bin/python2#" -i tools/run_tests/python_utils/antagonist.py
```

Comment:
Nothing to fix here, but this makes me think we should make fix_interpreter work with build_deps 😞

Reply:
Yes this is a common thing I’m seeing.  I’ll try to write up a feature request for this.